pkg/provision/workspace/deployment_provisioning.go

```diff
@@ -37,17 +37,17 @@ var infrastructureFailures = []string{
 // DetermineProvisioningFailureReason scans a deployment provisioning status info message
 // and returns the corresponding failure reason.
 // If a failure reason cannot be found, an Unknown reason is returned.
-func DetermineProvisioningFailureReason(status workspace.DeploymentProvisioningStatus) FailureReason {
+func DetermineProvisioningFailureReason(status DeploymentProvisioningStatus) metrics.FailureReason {
 	failStartupMsg := status.Info()
```

Comment:
Avoiding the import cycle:

```suggestion
func DetermineProvisioningFailureReason(failStartupMsg string) metrics.FailureReason {
```

Then change all call-sites to `metrics.DetermineProvisioningFailureReason(deploymentStatus.Info())`